### PR TITLE
feat: add `micro` completion spec

### PR DESCRIPTION
**Pull request by t-bre**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**
Add completion spec for [micro](https://github.com/zyedidia/micro) editor.

**Additional info:**

### CODE REVIEW

File: src/micro.ts [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+            name: "plugin",
+            isVariadic: true,
+          },
+        },
+        {
+          name: "search",
+          description: "Search for a plugin",
+          args: {
+            name: "plugin",
+            isVariadic: false,
```
Comment (fedeci) [inline src/micro.ts:45]: This is implicit.
```suggestion
```

File: src/micro.ts [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+    {
+      name: "--clean",
+      description: "Cleans the configuration directory",
+    },
+    {
+      name: "--config-dir",
+      description: "Specify a custom location for the configuration directory",
+      args: {
+        name: "dir",
+        isVariadic: false,
```
Comment (fedeci) [inline src/micro.ts:69]: Same here

**Review by fedeci — CHANGES_REQUESTED** [review]

**Review by t-bre — COMMENTED** [review]
Removed implicit `isVariadic: false`

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**